analyze.py

```diff
@@ -143,14 +191,54 @@ def _serialize_json(self, data):
        return json.dumps(data, indent=4)
    
     def _serialize_csv(self, data):
-       return json.dumps(data, indent=3) #fixme - make this export csv instead of json
-
+        with open('output.csv', 'w', newline='') as csvfile:
+            csvwriter = csv.writer(csvfile, delimiter=',',quotechar=',', quoting=csv.QUOTE_NONE,escapechar="\\")	
+            csvwriter.writerow(['']+['students_per_category'])
+            csvwriter.writerow(['']+['round_1']+['round_2']+['round_3'])
+            SpC = analayzer.worksheet["students_per_category"] #SpC = Students per category
+            categories = analayzer.category_amount
+            for i in range(categories):
+                csvwriter.writerow(['category_'+str(i+1)]+[str(SpC[0][i])]+[str(SpC[1][i])]+[str(SpC[2][i])])
+            h1=['']*7+["round_1"]+['']*43+["round_2"]+['']*43+["round_3"]+['']*43
+            roundstring1 = ['google_result_rating']+['']*19+['selections_per_category']+['']*9+['']+['average_google_category_rank']+['']*12
+            h2=['']*4+["jdistances"]+['']*2+roundstring1*3
+            roundstring2 = [str(i+1) for i in range(20)]+[str(i+1) for i in range(10)]+['jdistance']+[str(i+1) for i in range(10)]+['pearson_pvalue']+['pearson_coefficient']+["number_of_categories_selected"]
+            h3=["student_id"]+["student_last_name"]+["student_name"]+["student_notes"]+['rounds_1&2']+['rounds_1&3']+['rounds_2&3']+roundstring2*3
+            csvwriter.writerow([''])
+            csvwriter.writerow(h1)
+            csvwriter.writerow(h2)
+            csvwriter.writerow(h3)
+            students=analayzer.worksheet["students"]
+
+            for t in students:    
+                line=[]
+                snotes='' #problem stopping snotes from seperating lines
+                #for temp in t['student_notes'].strip().split('\n'): snotes+=temp
+                line+=[str(t['student_id'])]+[str(t['student_last_name'])]+[str(t['student_name'])]+[snotes]
+                temp = t['jdistances']
```

Comment:
will make it more clear by using proper names in separate methods after fixing the two major bugs.
again... sorry :\